model/api/rest_app.py

```diff
@@ -318,6 +324,82 @@ def create_resource(
         return resource
 
 
+@router.post(
+    "/collections/{collection_id}/resources/urls", status_code=201, tags=["resources"]
+)
+async def create_resource_from_url_list(
+    collection_id: UUID,
+    url_list: UrlListBase,
+    session: Annotated[Session, Depends(get_session)],
+    user: Annotated[User, Depends(get_current_user)],
+) -> list[Resource]:
+    """
+    Endpoint to upload a file to a specified collection.
+
+    Args:
+        session: DB session
+        user: The logged-in user from auth JWT or None
+        collection_id (str): The collection to upload the file to.
+        url_list (UrlListBase): The urls being uploaded.
+
+    Returns:
+        Resources
+    """
+    __check_user_is_member_of_collection(user, collection_id, session)
+
+    urls = url_list.urls
+    for url in urls:
+        parsed = urlparse(url)
+        if not parsed.scheme or not parsed.netloc:
+            raise HTTPException(
+                status_code=422, detail="Unsupported URLs found in URL list"
```

Comment:
Should we tell them which URL it was? Also we can't be sure it was multiple as this exception will fly on the first unparseable URL. Sth like "Unsupported URL: {url}" might be clearer